File: LeetCode/Simulation/Python/Leetcode1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        n, st, cnt = len(s) // 2, ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'], 0
        for i in range(n):
            if s[i] in st: cnt += 1
        for i in range(n, n * 2):
            if s[i] in st: cnt -= 1
            if cnt < 0: return False
        return cnt == 0

File: LeetCode/Simulation/Rust/Leetcode1704-determine-if-string-halves-are-alike.rs
use std::collections::HashSet;
impl Solution {
  pub fn halves_are_alike(s: String) -> bool {
    let (n, mut cnt) = (s.len() / 2, 0);
    let st = HashSet::from([b'a', b'e', b'i', b'o', b'u', b'A', b'E', b'I', b'O', b'U']);
    for i in 0..n {
      if st.contains(&s.as_bytes()[i]) {
        cnt += 1;
      }
    }
    for i in n..n * 2 {
      if st.contains(&s.as_bytes()[i]) {
        cnt -= 1;
      }
      if cnt < 0 {
        return false;
      }
    }
    cnt == 0
  }
}
